### PR TITLE
Document volume plugin options and allow config file options for photon driver.

### PR DESCRIPTION
**Pull request by govint**
1. Updated documentation of supported config file options.
2. Modified main.go to use the config file for photon driver options.

Verified error handling with command line options for photon driver, will be testing config file based options as well.

### CODE REVIEW

File: docs/user-guide/docker-plugin-drivers.md [hunk 1]
```diff
@@ -13,3 +13,16 @@ The Docker Vmdk volume driver supports provisioning and managing docker volumes
 The Docker Photon Volume driver supports provisioning and managing docker volumes on a Photon platform consisting of a cluster of ESX hosts managed via a Photon controller instance. Docker volumes are created and managed entirely via the open Photon platform API via the Photon controller.
 
 The Docker Volume plugin can support either or both types of volumes, as required, on a given Docker host.
+
+## Configuring the Docker Volume Plugin
+The docker volume plugin is designed to load run time options and values from a json configuration file (default /etc/docker-volume-vsphere.conf) on the host. The user can also provide a configuration file, via the "--config" option, specifying the full path of the file. The file contains the values for run time options used by the plugin. Options that are currently recognized include the below set. Options provided on the command line will override those in the configuration file.
+
+# Options for the photon volume driver
```
Comment (kerneltime) [inline docs/user-guide/docker-plugin-drivers.md]: This is not enough, either package a default config that can be renamed to /etc/docker-volume-vsphere.conf or include the sample JSON here at a minimum.

Comment (govint) [inline docs/user-guide/docker-plugin-drivers.md]: Added a sample configuration.

File: docs/user-guide/docker-plugin-drivers.md [hunk 2]
```diff
@@ -13,3 +13,28 @@ The Docker Vmdk volume driver supports provisioning and managing docker volumes
...
+Target    - URL at which to contact the Photon Controller
```
Comment (pdhamdhere) [inline docs/user-guide/docker-plugin-drivers.md]: Is this API Endpoint? May be add "for e.g. http://...."

File: docs/user-guide/docker-plugin-drivers.md [hunk 3]
```diff
@@ -13,3 +13,28 @@ The Docker Vmdk volume driver supports provisioning and managing docker volumes
...
+Project   - project ID in Photon to which the docker host belongs
```
Comment (pdhamdhere) [inline docs/user-guide/docker-plugin-drivers.md]: Can we add commands or link to Photon documentation on how to to get "Project" and "Host" details. Also add sample value?

File: docs/user-guide/docker-plugin-drivers.md (hunk 2 above)
Comment (govint) [inline docs/user-guide/docker-plugin-drivers.md]: Added format for target URL.

File: docs/user-guide/docker-plugin-drivers.md (hunk 3 above)
Comment (govint) [inline docs/user-guide/docker-plugin-drivers.md]: Added description of the  project and host IDs and link to photon CLI.

File: docs/user-guide/docker-plugin-drivers.md [hunk 4]
```diff
@@ -13,3 +13,33 @@ The Docker Vmdk volume driver supports provisioning and managing docker volumes
...
+MaxLogSizeMb  - max. size of the plugin log file
+MaxLogAgeDays - number of days to retain plugin log files
+
+## Sample plugin configuration
+{
+	"MaxLogAgeDays": 28,
+	"MaxLogSizeMb": 100,
+	"LogPath": "/var/log/docker-volume-vsphere.log",
+	"LogLevel": "info"
+	"Target" : "http://<photon_controller_ip>:<target port>"
```
Comment (pdhamdhere) [inline docs/user-guide/docker-plugin-drivers.md]: Add comment "Following options only apply to Photon Platform".

Comment (govint) [inline docs/user-guide/docker-plugin-drivers.md]: Done.

**Pull request merged**